src/com/amazon/corretto/crypto/provider/Loader.java

```diff
@@ -429,6 +430,33 @@ private static byte[] bootstrapRng(int numBytes) throws IOException {
     throw new AssertionError("Unable to read enough entropy");
   }
 
+  private static void verifyDirectoryOwnershipAndPermissions(Path directory) throws IOException {
+    UserPrincipal directoryOwner = Files.getOwner(directory);
+    String currentUsername = System.getProperty("user.name");
```

Comment:
is this sufficiently robust? JVM system properties can be overridden. i don't have a better idea than shelling out to `whoami` or equivalent, but that's not particularly portable...

so, not a blocker, just something for us to hopefully discuss a bit.